Formula/uptoc.rb

```diff
@@ -0,0 +1,25 @@
+# This file was generated by GoReleaser. DO NOT EDIT.
+class Uptoc < Formula
+  desc "Convenient static file deployment tool that supports multiple platforms"
+  homepage "https://github.com/saltbo/uptoc"
+  version "1.4.1"
+  bottle :unneeded
+
+  if OS.mac?
+    url "https://github.com/saltbo/uptoc/releases/download/v1.4.1/uptoc-macos-amd64.tar.gz"
+    sha256 "1a81f1d31b9b3eedbb9bc89c708089b81f543a26d58aea46611b52679d7ddb32"
+  elsif OS.linux?
+    if Hardware::CPU.intel?
+      url "https://github.com/saltbo/uptoc/releases/download/v1.4.1/uptoc-linux-amd64.tar.gz"
+      sha256 "efc75ad1dfeea44f6cbdd416c654dce9768e866ffbcd6a63e829d67015b6c95b"
+    end
+  end
+
+  def install
+    bin.install "uptoc"
+  end
+
+  test do
+    system "#{bin}/uptoc", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.